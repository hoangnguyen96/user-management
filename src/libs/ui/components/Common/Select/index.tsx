import {
  MenuItem,
  Select as SelectBase,
  Stack,
  Typography,
} from "@mui/material";
import { themeDefault } from "@app/ui/themes";
import { UserResponse } from "@app/models";

interface SelectProps {
  value?: string;
  errorMessage?: string;
  isInvalid?: boolean;
  listUser?: UserResponse[];
  onChange?: (event: any) => void;
}

export const Select = ({
  value,
  errorMessage,
  isInvalid,
  listUser,
  onChange,
  ...rest
}: SelectProps) => {
  return (
    <Stack width="200px" mb={errorMessage ? 0 : "20px"}>
      <SelectBase
        value={value}
        displayEmpty
        onChange={onChange}
        inputProps={{ "aria-label": "Without label" }}
        sx={{
          border: "unset",
          maxWidth: "200px",
          borderRadius: "12px",
          "& .MuiOutlinedInput-notchedOutline": {
            borderColor:
              errorMessage || isInvalid
                ? themeDefault.palette.error.main
                : "#b5b7C0",
          },
          "&:hover .MuiOutlinedInput-notchedOutline": {
            borderColor:
              errorMessage || isInvalid
                ? themeDefault.palette.error.main
                : themeDefault.palette.primary.main,
          },
          "&:focus .MuiOutlinedInput-notchedOutline": {
            borderColor:
              errorMessage || isInvalid
                ? themeDefault.palette.error.main
                : themeDefault.palette.primary.main,
          },
        }}
        {...rest}
      >
        {listUser?.map(({ id, fullName }) => (
          <MenuItem key={id} value={id}>
            {fullName}
          </MenuItem>
        ))}
      </SelectBase>
      {errorMessage && (
        <Typography
          variant="caption"
          fontSize="12px"
          ml="10px"
          color={themeDefault.palette.error.main}
        >
          {errorMessage}
        </Typography>
      )}
    </Stack>
  );
};
