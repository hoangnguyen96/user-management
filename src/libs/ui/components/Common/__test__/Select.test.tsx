import { render } from "@testing-library/react";

// Component
import { Select } from "../Select";

describe("Select component", () => {
  it("Should render snapshot correctly", () => {
    expect(render(<Select />)).toMatchSnapshot();
  });
});
