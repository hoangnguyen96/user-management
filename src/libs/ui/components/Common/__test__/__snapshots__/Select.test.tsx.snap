// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-1f0fbh5-MuiStack-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-jvh0ve-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Without label"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-1f0fbh5-MuiStack-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-jvh0ve-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-controls=":r0:"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Without label"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="combobox"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
